### ISSUE TITLE
Active object type after killing/deleting

### ISSUE THREAD

**Issue by WhatisRT** [clemera/objed#38]
After killing or deleting an object, the active object seems to switch to either the `char` or the `line` object, depending on what object has been selected before. I think it would be more natural if the object type would stay the same, which would make it really easy to kill a couple of objects quickly (and without knowing before I hit the `k` button how many objects I want to kill).
This seems like a general principle: If the user uses a command which doesn't explicitly have the purpose of changing the type of object selected, the command should try its best to have the same type of object active afterwards, because there is no reason to think the user wants a different object type active.

**Comment by clemera** [comment]
I just thought about this yesterday :) The problem is that after killing an object like a word you don't want you point to jump to the next highlighted word, it should stay where it is and a repeated kill should kill the whitespace after your current position AND the next word. And another questions is what should be highlighted. You would probably want to see the whitespace AND the word highlighted, because that's what would be killed. With other objects you probably want different behavior, for example after killing a string you probably don't want kill everything after the current string plust the next string.

**Comment by WhatisRT** [comment]
I agree with everything you said, and there is even the extra issue of direction here: do I want to kill towards the left or right? It may or may not be a good idea to determine that from the cursor position in relation to the object: it the cursor is closer the end of a word, one probably wants to kill towards the left, while one should kill towards the right otherwise.
As for killing stuff in between, I'm currently unaware of the implementation details, but I assume text objects can be assigned some properties? In that case, one could have a property that controls what to do, maybe with the options: kill whitespace, kill everything, kill nothing. One could also have a property what text object to activate after some object is killed, for better flexibility. If it is `nil`, stay with the current object type, otherwise activate whatever is mentioned there. I could for example imagine that depending on what people are using, it might be more sensible to select a sexp after a string has been killed.

**Comment by clemera** [comment]
I think using properties which decide the behavior based on object type are the right idea, but first I have to figure out how to implement such exit behavior in general. I will start implementing the behavior for words and go from there.

**Comment by clemera** [comment]
The direction should default to forward in my opinion it would be confusing to decide the direction on a case by case basis. But maybe one could press a key to change the direction.

**Comment by clemera** [comment]
I have added a continuation object feature now. For operations like comment operation you wouldn't like to keep the current object type active, so the feature is only used if the operation killed/deleted the object contents from the buffer. I played a bit with it and I really like the new behavior. Please give it some testing and report any issues you have with it.

**Comment by WhatisRT** [comment]
Works great on the first impression!

In regards to the direction, one could do it vim style: lower case for forwards, upper case for backwards. That's probably the easiest way.

**Comment by clemera** [comment]
> do it vim style: lower case for forwards, upper case for backwards

That's a great suggestion. I have implemented it for "K" and "D". Currently killing backwards has some issues though: when pasting back the killed text is reversed...

**Comment by clemera** [comment]
Ok the backward kill works now correctly, for some weird reason I don't understand currently passing the markers directly failed the prepend/append check in `kill-append`.

**Comment by WhatisRT** [comment]
Seems to work great! Now the only thing I'm missing from this is for those functions to accept prefix arguments...

**Comment by clemera** [comment]
Prefix args should work now for kill and delete, too.

**Comment by WhatisRT** [comment]
Awesome! But now `undo` is a bit broken: `3k` needs 3 `undo` commands to be undone.

**Comment by clemera** [comment]
Hm, this isn't true for me. Although it makes sense, might be some problem with my Emacs config, I have to investigate. What Emacs version do you use by the way?

**Comment by WhatisRT** [comment]
I'm on `26.1`, running spacemacs develop, but with evil disabled.

**Comment by clemera** [comment]
Ok, I don't know what's wrong with my config but I get the same behavior if I install objed in a fresh instance so I removed the additional undo boundaries again.

**Comment by WhatisRT** [comment]
Great, works for me now!

**Comment by clemera** [comment]
I would like to leave that open, to remind me finishing the working but still not final implementation I would like to have. Thanks for the great suggestion!
